src/libcore/cmp.rs

```diff
@@ -567,6 +568,33 @@ pub trait Ord: Eq + PartialOrd<Self> {
     where Self: Sized {
         if self <= other { self } else { other }
     }
+     
+    /// Returns max if self is greater than max, and min if self is less than min.
+    /// Otherwise this will return self.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(clamp)]
+    ///
+    /// assert!((-3).clamp(-2..=1) == -2);
+    /// assert!(0.clamp(-2..=1) == 0);
+    /// assert!(2.clamp(-2..=1) == 1);
+    /// ```
+    ///
+    /// # Panics
+    /// Panics if range.is_empty().
+    #[unstable(feature = "clamp", issue = "44095")]
+    #[inline]
+    fn clamp(self, range : RangeInclusive<Self>) -> Self
```

Comment:
Why does this only work with RangeInclusive?

Reply:
Because to work with a `Range` would require a `predecessor` method on `Self` to be able to generate a value inside the range.  (Now, whether that means it should take a `RangeInclusive` or just two values is another question -- https://github.com/rust-lang/rfcs/pull/1961#issuecomment-302600351)

Reply:
It seems like this should just take 2 arguments then.

Reply:
That makes sense, I will rewrite it to the original 2 arguments